src/T2DMap.cpp

```diff
@@ -4257,13 +4262,29 @@ void T2DMap::slot_setArea()
         arealist_combobox->addItem(qsl("%1 (%2)").arg(sortedAreaList.at(i), QString::number(areaId)), QString::number(areaId));
     }
 
-
-
     if (set_room_area_dialog->exec() == QDialog::Rejected) { // Don't proceed if "cancel" was pressed
         return;
     }
 
-    int newAreaId = arealist_combobox->itemData(arealist_combobox->currentIndex()).toInt();
+    auto newArea = arealist_combobox->itemData(arealist_combobox->currentIndex());
+    int  newAreaId;
+    if (newArea.isValid()) {
+        newAreaId = newArea.toInt();
+    } else if (sortedAreaList.contains(arealist_combobox->currentText().trimmed())) {
+        newAreaId = mpMap->mpRoomDB->getAreaNamesMap().key(arealist_combobox->currentText());
+    } else {
+        auto newAreaName = arealist_combobox->currentText();
+        newAreaId = mpMap->mpRoomDB->addArea(newAreaName);
+        if (newAreaId == 0) {
+            return; // This should not happen, but if it fails addArea underneath will log reason
+        }
+        mpMap->postMessage(tr("[  OK  ]  - Added %1 (%2) area to map.").arg(newAreaName, QString::number(newAreaId)));
+        mpMap->mUnsavedMap = true;
+
+        if (mpMap->mpMapper) {
+            mpMap->mpMapper->updateAreaComboBox();
+        }
```

Comment:
The check *seems* unnecessary to me - the `T2DMap` class is used/created *by* what `mpMap->mpMapper` (a `dlgMapper` class instance) is pointing to - and one could not get to this point in the application's execution without it pointing to a valid thing - can it? So I think it should be okay to have:
```suggestion
        mpMap->mpMapper->updateAreaComboBox();
```